### DISCUSSION TITLE
How can I keep the original file extension?

### DISCUSSION THREAD

**Discussion by Raxel21** [webdiscus/html-bundler-webpack-plugin#3] [Q&A]
Hi, I hope everything is going well for you, today I discovered your plugin and I find it very useful.

Well, I am having a problem working with `.php`, with `.html` it works perfectly. I tried to adapt the code but I did not succeed. 

I tried 2 ways

It works, but it changes the extension to `.html`, so it breaks the PHP code.
```javascript
new HtmlBundlerPlugin({
  test: /\.php$/i,
  verbose: true,
  entry: {
    index: path.join(__dirname, 'src', 'public', 'index.php'),
  },
  js: {
    filename: '[name].[contenthash:8].js',
  },
  css: {
    filename: '[name].[contenthash:8].css',
  },
})
```
So I tried this other way 

It also works, but it keeps changing the extension to `.html`, but this time I get an error:
`html-bundler-webpack-plugin: Failed to execute the template function`
I thought that just returning the template in the `preprocessor` line would work but I get the above error.
```javascript
new HtmlBundlerPlugin({
  test: /\.php$/i,
  verbose: true,
  entry: {
    home: {
      import: path.join(__dirname, 'src', 'public', 'index.php'),
      filename: 'index.php',
    },
  },
  js: {
    filename: '[name].[contenthash:8].js',
  },
  css: {
    filename: '[name].[contenthash:8].css',
  },
  loaderOptions: {
    preprocessor: (template, { data }) => template,
  },
})
``` 
Any suggestions please. Or maybe I forgot something from the documentation?
**Thank you in advance**

**Comment by webdiscus** [comment]
Hello @Raxel21 

thanks for the interesting use case.
I will try to reproduce the problem and find a solution.

**Reply by Raxel21** [reply]
**Thank you in advance!**
